### ISSUE TITLE
Test failure with Pyhon 3.8

### ISSUE THREAD

**Issue by FRidh** [jaraco/path.py#172]
```
============================= test session starts ==============================
platform linux -- Python 3.8.0, pytest-5.2.1, py-1.8.0, pluggy-0.13.0 -- /nix/store/n483rgwmx33wd0di0dqhsfr5g291z5fr-python3-3.8.0/bin/python3.8
cachedir: .pytest_cache
rootdir: /build/path.py-12.0.1, inifile: pytest.ini
plugins: flake8-1.0.4
collecting ... collected 80 items / 1 deselected / 79 selected

setup.py::FLAKE8 PASSED                                                  [  1%]
test_path.py::FLAKE8 PASSED                                              [  2%]
test_path.py::TestBasics::test_relpath[Path] PASSED                      [  3%]
test_path.py::TestBasics::test_construction_from_none[Path] PASSED       [  5%]
test_path.py::TestBasics::test_construction_from_int[Path] PASSED        [  6%]
test_path.py::TestBasics::test_string_compatibility[Path] PASSED         [  7%]
test_path.py::TestBasics::test_properties[Path] PASSED                   [  8%]
test_path.py::TestBasics::test_methods[Path] PASSED                      [ 10%]
test_path.py::TestBasics::test_UNC[Path] PASSED                          [ 11%]
test_path.py::TestBasics::test_explicit_module[Path] PASSED              [ 12%]
test_path.py::TestBasics::test_explicit_module_classes[Path] PASSED      [ 13%]
test_path.py::TestBasics::test_joinpath_on_instance[Path] PASSED         [ 15%]
test_path.py::TestBasics::test_joinpath_to_nothing[Path] PASSED          [ 16%]
test_path.py::TestBasics::test_joinpath_on_class[Path] PASSED            [ 17%]
test_path.py::TestBasics::test_joinpath_fails_on_empty[Path] PASSED      [ 18%]
test_path.py::TestBasics::test_joinpath_returns_same_type[Path] PASSED   [ 20%]
test_path.py::TestSelfReturn::test_makedirs_p[Path] PASSED               [ 21%]
test_path.py::TestSelfReturn::test_makedirs_p_extant[Path] PASSED        [ 22%]
test_path.py::TestSelfReturn::test_rename[Path] PASSED                   [ 24%]
test_path.py::TestSelfReturn::test_mkdir[Path] PASSED                    [ 25%]
test_path.py::TestSelfReturn::test_touch[Path] PASSED                    [ 26%]
test_path.py::TestScratchDir::test_context_manager[Path] PASSED          [ 27%]
test_path.py::TestScratchDir::test_touch[Path] PASSED                    [ 29%]
test_path.py::TestScratchDir::test_listing[Path] PASSED                  [ 30%]
test_path.py::TestScratchDir::test_listdir_other_encoding[Path] PASSED   [ 31%]
test_path.py::TestScratchDir::test_makedirs[Path] PASSED                 [ 32%]
test_path.py::TestScratchDir::test_shutil[Path] PASSED                   [ 34%]
test_path.py::TestScratchDir::test_patterns[Path] PASSED                 [ 35%]
test_path.py::TestScratchDir::test_unicode[Path] PASSED                  [ 36%]
test_path.py::TestScratchDir::test_chunks[Path] PASSED                   [ 37%]
test_path.py::TestScratchDir::test_samefile[Path] PASSED                 [ 39%]
test_path.py::TestScratchDir::test_rmtree_p[Path] PASSED                 [ 40%]
test_path.py::TestScratchDir::test_rmdir_p_exists[Path] PASSED           [ 41%]
test_path.py::TestScratchDir::test_rmdir_p_nonexistent[Path] PASSED      [ 43%]
test_path.py::TestMergeTree::test_with_nonexisting_dst_kwargs[Path] PASSED [ 44%]
test_path.py::TestMergeTree::test_with_nonexisting_dst_args[Path] PASSED [ 45%]
test_path.py::TestMergeTree::test_with_existing_dst[Path] PASSED         [ 46%]
test_path.py::TestMergeTree::test_copytree_parameters[Path] PASSED       [ 48%]
test_path.py::TestMergeTree::test_only_newer[Path] PASSED                [ 49%]
test_path.py::TestChdir::test_chdir_or_cd[Path] PASSED                   [ 50%]
test_path.py::TestSubclass::test_subclass_produces_same_class[Path] PASSED [ 51%]
test_path.py::TestTempDir::test_constructor[Path] PASSED                 [ 53%]
test_path.py::TestTempDir::test_next_class[Path] PASSED                  [ 54%]
test_path.py::TestTempDir::test_context_manager[Path] PASSED             [ 55%]
test_path.py::TestTempDir::test_context_manager_exception[Path] PASSED   [ 56%]
test_path.py::TestTempDir::test_context_manager_using_with[Path] PASSED  [ 58%]
test_path.py::TestUnicode::test_walkdirs_with_unicode_name[Path] PASSED  [ 59%]
test_path.py::TestPatternMatching::test_fnmatch_simple[Path] PASSED      [ 60%]
test_path.py::TestPatternMatching::test_fnmatch_custom_mod[Path] PASSED  [ 62%]
test_path.py::TestPatternMatching::test_fnmatch_custom_normcase[Path] PASSED [ 63%]
test_path.py::TestPatternMatching::test_listdir_simple[Path] PASSED      [ 64%]
test_path.py::TestPatternMatching::test_listdir_empty_pattern[Path] PASSED [ 65%]
test_path.py::TestPatternMatching::test_listdir_patterns[Path] PASSED    [ 67%]
test_path.py::TestPatternMatching::test_listdir_custom_module[Path] PASSED [ 68%]
test_path.py::TestPatternMatching::test_listdir_case_insensitive[Path] PASSED [ 69%]
test_path.py::TestPatternMatching::test_walk_case_insensitive[Path] PASSED [ 70%]
test_path.py::TestInPlace::test_line_by_line_rewrite[Path] PASSED        [ 72%]
test_path.py::TestInPlace::test_exception_in_context[Path] PASSED        [ 73%]
test_path.py::TestSpecialPaths::test_basic_paths[Path] SKIPPED           [ 74%]
test_path.py::TestSpecialPaths::test_unix_paths[Path] SKIPPED            [ 75%]
test_path.py::TestSpecialPaths::test_unix_paths_fallback[Path] SKIPPED   [ 77%]
test_path.py::TestSpecialPaths::test_property[Path] SKIPPED              [ 78%]
test_path.py::TestSpecialPaths::test_other_parameters[Path] SKIPPED      [ 79%]
test_path.py::TestSpecialPaths::test_multipath[Path] SKIPPED             [ 81%]
test_path.py::TestSpecialPaths::test_reused_SpecialResolver[Path] SKIPPED [ 82%]
test_path.py::TestMultiPath::test_for_class[Path] PASSED                 [ 83%]
test_path.py::TestMultiPath::test_detect_no_pathsep[Path] PASSED         [ 84%]
test_path.py::TestMultiPath::test_detect_with_pathsep[Path] PASSED       [ 86%]
test_path.py::TestMultiPath::test_iteration[Path] PASSED                 [ 87%]
test_path.py::test_no_dependencies[Path] PASSED                          [ 88%]
test_path.py::test_version[Path] FAILED                                  [ 89%]
docs/conf.py::FLAKE8 PASSED                                              [ 91%]
path/__init__.py::FLAKE8 PASSED                                          [ 92%]
path/__init__.py::path.Path.chunks PASSED                                [ 93%]
path/__init__.py::path.Path.stem PASSED                                  [ 94%]
path/__init__.py::path.Path.with_suffix PASSED                           [ 96%]
path/__init__.py::path._multi_permission_mask PASSED                     [ 97%]
path/__init__.py::path._permission_mask PASSED                           [ 98%]
path/matchers.py::FLAKE8 PASSED                                          [100%]

=================================== FAILURES ===================================
______________________________ test_version[Path] ______________________________

    def test_version():
        """
        Under normal circumstances, path should present a
        __version__.
        """
>       assert re.match(r'\d+\.\d+.*', path.__version__)
E       AssertionError: assert None
E        +  where None = <function match at 0x7fffef99edc0>('\\d+\\.\\d+.*', 'unknown')
E        +    where <function match at 0x7fffef99edc0> = re.match
E        +    and   'unknown' = path.__version__

test_path.py:1254: AssertionError
=============================== warnings summary ===============================
test_path.py::test_version[Path]
  /build/path.py-12.0.1/test_path.py:1254: PytestAssertRewriteWarning: asserting the value None, please use "assert is None"
    assert re.match(r'\d+\.\d+.*', path.__version__)

-- Docs: https://docs.pytest.org/en/latest/warnings.html
====== 1 failed, 71 passed, 7 skipped, 1 deselected, 1 warnings in 7.78s =======
```

**Comment by ssbarnea** [comment]
It worth mentioning that py38 is already supported by travis, without any hacks.

**Comment by jaraco** [comment]
Tests are passing on Python 3.8 now, so I'm guessing this issue was temporary or will be resolved with #174.
